src/task/collection.coffee

```diff
@@ -67,6 +81,7 @@ getModuleInfo = (taskId, cnxUrl = '') ->
   moduleInfo
 
 api.channel.on("task.*.receive.*", update)
+api.channel.on('task.*.receive.failure', checkFailure)
 
```

Comment:
Sorry, that was unclear.. Yeah, I get why it's good to have the `api.channel.on('task.*.receive.failure', checkFailure)`.  My concern was that `update` will fail if it's called for an server error that doesn't have any eventData, such as a `500`  

I think `update` would need to be modified with `?` guards, or only listen to a more specific wildcard that would exclude `.failure`